compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Dar.hs

```diff
@@ -105,80 +106,100 @@ buildDar service file mbExposedModules pkgName sdkVersion buildDataFiles dalfInp
               "The following modules are declared in exposed-modules but are not part of the DALF: " <>
               show (S.toList missingExposed)
       let dalf = encodeArchiveLazy pkg
+      let hash = T.unpack $ encodePackageHash pkg
       -- get all dalf dependencies.
       dalfDependencies0 <- getDalfDependencies file
       let dalfDependencies =
               [ (T.pack $ unitIdString unitId, dalfPackageBytes pkg) | (unitId, pkg) <- Map.toList dalfDependencies0 ]
       -- get all file dependencies
       fileDependencies <- MaybeT $ getDependencies file
+      let dataFiles = buildDataFiles (hash, pkg)
       liftIO $
         createArchive
           dalf
+          hash
           (toNormalizedFilePath $ takeDirectory file')
           dalfDependencies
           (file:fileDependencies)
-          (buildDataFiles pkg)
+          dataFiles
           pkgName
           sdkVersion
 
 -- | Helper to bundle up all files into a DAR.
-createArchive
-  :: BSL.ByteString -- ^ DALF
-  -> NormalizedFilePath -- ^ Module root used to locate interface and source files
-  -> [(T.Text, BS.ByteString)] -- ^ DALF dependencies
-  -> [NormalizedFilePath] -- ^ Module dependencies
-  -> [(String, BS.ByteString)] -- ^ Data files
-  -> String -- ^ Name of the main DALF
-  -> String -- ^ SDK version
-  -> IO BS.ByteString
-createArchive dalf modRoot dalfDependencies fileDependencies dataFiles name sdkVersion = do
+createArchive ::
+       BSL.ByteString -- ^ DALF
+    -> String -- ^ Package id
+    -> NormalizedFilePath -- ^ Module root used to locate interface and source files
+    -> [(T.Text, BS.ByteString)] -- ^ DALF dependencies
+    -> [NormalizedFilePath] -- ^ Module dependencies
+    -> [(String, BS.ByteString)] -- ^ Data files
+    -> String -- ^ Name of the main DALF
+    -> String -- ^ SDK version
+    -> IO BS.ByteString
+createArchive dalf hash modRoot dalfDependencies fileDependencies dataFiles name sdkVersion
```

Comment:
What has actually changed in this signature? Can we please undo the formatting change.

Reply:
Nothing, but let's keep the formatting change, the way it's formatted right now makes my eyes bleed.